bt_ddos_shield/miner_shield.py

```diff
@@ -1,74 +1,481 @@
+        self._handle_changed_validators(changed_validators)
+
+        if deprecated_validators or new_validators or changed_validators:
+            # if anything changed update manifest file and publish new version to blockchain
+            self._add_task(MinerShieldUpdateManifestTask())
+
+    def _handle_ban_validator(self, validator_hotkey: Hotkey):
+        """
+        Ban validator by its hotkey. If something changed, MinerShieldValidatorsChangedTask will apply asynchronously
+        this change where needed.
+        """
+        self.state_manager.add_banned_validator(validator_hotkey)
+        self._event("Validator {validator_hotkey} added to banned set", validator_hotkey=validator_hotkey)
+        self._add_task(MinerShieldValidatorsChangedTask())
 
-        Args:
-            validator_hotkey: The hotkey of the validator.
+    def _handle_update_manifest(self):
+        """
+        Update manifest file and schedule publishing it to blockchain.
+        """
+        current_state: MinerShieldState = self.state_manager.get_state()
+        manifest: Manifest = self.manifest_manager.create_manifest(current_state.validators_addresses,
+                                                                   current_state.known_validators)
+        address: Address = self.manifest_manager.upload_manifest(manifest)
+        self.state_manager.set_manifest_address(address)
+        self._event("Manifest updated, new address: {address}", address=address)
+        self._add_task(MinerShieldPublishManifestTask())
+
+    def _handle_publish_manifest(self):
+        """
+        Publish info about current manifest file to blockchain if it is not already there.
+        """
+        expected_address: Address = self.state_manager.get_state().manifest_address
+        current_address: Address = self.blockchain_manager.get_address(self.miner_hotkey)
+        if current_address == expected_address:
+            self._event("Manifest address already published")
+        else:
+            self.blockchain_manager.put_address(self.miner_hotkey, expected_address)
+            self._event("Manifest published")
+
+    def _event(self, template: str, exception: Exception = None, **kwargs):
+        return self.event_processor.event(template, exception, **kwargs)
+
+
+class AbstractMinerShieldTask(ABC):
+    """
+    Task to be executed by shield worker.
+    """
+
+    NAME_DELETER = re.compile(r'^MinerShield(.*)Task$')
+
+    @abstractmethod
+    def run(self, miner_shield: MinerShield):
+        """
+        Run task in miner_shield context.
         """
         pass
+
+    def __repr__(self):
+        return self.NAME_DELETER.sub(r'\1', self.__class__.__name__)
+
+
+class MinerShieldInitializeTask(AbstractMinerShieldTask):
+    def run(self, miner_shield: MinerShield):
+        # noinspection PyProtectedMember
+        miner_shield._handle_initialize()
+
+
+class MinerShieldDisableTask(AbstractMinerShieldTask):
```

Comment:
insteaf of define these as classes - the functions could just be passed directly

Reply:
Hmm, I'm not sure about this. Adding these tasks as class allows us to print their names in logs when handling - I know I can use method.__name__ for this but getting class name looks nicer for me. Also some tasks have params, so I will need to pass method and params to queue, so anyway I need to wrap them in some tuple or something - having class for this also looks nicer for me.